### PR TITLE
MRG: Add colors to sphere ConductorModel

### LINKED ISSUE #4723: ENH: plot_alignment with sphere model
Currently there is no way to determine the "up" direction when using a sphere model. This is fine from a forward modeling standpoint because it doesn't matter, but for checking alignment it actually matters because you can't see if the head is in a reasonable position (i.e., not upside-down). You can check this with `coord_frame='head'` but that's not an optimal solution -- it would be nice if we had some way of recognizing the "up" of that coordinate system.

Options/ideas:

1. make the sphere color non-uniform by e.g. making the top light gray and the bottom dark gray
2. make the sphere color non-uniform by marking the topmost face of the sphere with some different color (white?)
3. add optional 3-arrow axis indicators showing the origin each coordinate system (MRI, head, MEG)

Personally I like the idea of having (2) and (3) -- I'd find (3) interesting/cool for even proper BEM models, and it has some nice pedagogical value.

### PR DESCRIPTION
**Pull request by larsoner**
WIP because I need to 

- [x] ~~add colors to sphere~~
- [x] add terrain interaction
- [x] fix `setup_volume_source_space` to use innermost radius (instead of outermost)
- [x] update `sphere` docstring for `setup_volume_source_space`
- [x] add tests for `plot_alignment` and `setup_volume_source_space`
- [x] add axis indicators w / e.g. `axes=('meg', 'head', 'mri')`
- [x] update `whats_new`

Closes #4723.

### CODE REVIEW

**Comment by larsoner** [comment]
Progress thus far:

https://6349-1301584-gh.circle-artifacts.com/0/tmp/circle-artifacts.nkqgCXh/html/auto_tutorials/plot_source_alignment.html

**Comment by larsoner** [comment]
Okay this one is good to go from my end. I decided to scrap the coloring of the sphere -- it was hard to see, and adding axis indicators made things clear enough. I updated the tutorial to show the coordinate frames a bit for pedagogical purposes:

https://6355-1301584-gh.circle-artifacts.com/0/tmp/circle-artifacts.MspB6JX/html/auto_tutorials/plot_source_alignment.html

**Comment by larsoner** [comment]
@agramfort any time to look at this one? You mostly get to look at pictures and comment on their pedagogical / aesthetic value!

**Comment by larsoner** [comment]
@wmvanvliet you might also be interested in looking given you did some pedagogical stuff with source orientations / probably need to teach people about coordinate frames

@jona-sassenhagen IIRC we talked about coordinate frames at some point (?), feel free to look to see if this clarifies anything for you

**Comment by wmvanvliet** [comment]
Great to see our documentation getting so much love. Nice tutorial. If I could add one thing, it would be an explanation about which coordinate systems there are (head, device, mri), how they are defined and how they relate to each other. With such an explanation, we could then tell the user that `plot_transform` transforms everything into the "head" coordinate system, using the MRI->head and device->head transforms.

**Comment by larsoner** [comment]
> Nice tutorial. If I could add one thing, it would be an explanation about which coordinate systems there are (head, device, mri), how they are defined and how they relate to each other.

That list already exists here:

https://6355-1301584-gh.circle-artifacts.com/0/tmp/circle-artifacts.MspB6JX/html/auto_tutorials/plot_source_alignment.html#visualizing-coordinate-frames

Feel free to comment in-line on how to improve.

**Comment by larsoner** [comment]
Maybe bold the names to pull them out? Maybe move the section up somehow (though it would mess with the existing flow)?

**Comment by jona-sassenhagen** [comment]
Bolding would be good.

File: tutorials/plot_source_alignment.py [hunk 1]
```diff
@@ -66,33 +70,68 @@
...
-#           or a list of bem surfaces (:func:`mne.read_bem_surfaces`).
-sphere = mne.make_sphere_model(info=raw.info, r0='auto', head_radius='auto')
-mne.viz.plot_alignment(raw.info, subject='sample', eeg='projected',
-                       meg='helmet', bem=sphere, dig=True,
-                       surfaces=['brain', 'inner_skull', 'outer_skull',
-                                 'outer_skin'])
+# From this plot, where we have removed surfaces to more clearly see the
+# coordinate frame axes, you can see:
+#
+# 1. The Neuromage head coordinate frame that MNE uses (pink) is defined by
```
Comment (jona-sassenhagen) [inline tutorials/plot_source_alignment.py]: `Neuromag`?

File: tutorials/plot_source_alignment.py [hunk 2]
```diff
@@ -66,33 +70,68 @@
...
+#    the intersection of the line between the LPA and RPA, and the normal from
```
Comment (jona-sassenhagen) [inline tutorials/plot_source_alignment.py]: I don't know what "the normal from x to y" means.

File: tutorials/plot_source_alignment.py [hunk 3]
```diff
@@ -66,33 +70,68 @@
...
-# The previous is possible if you have the surfaces available from Freesurfer.
-# The function automatically searches for the correct surfaces from the
-# provided ``subjects_dir``. Otherwise, one option is to use a spherical
-# conductor model. It is passed through ``bem`` parameter.
+# Visualizing coordinate frames
+# -----------------------------
+# If you are curious about the origin and orientation of each of these
+# coordinate frames, you can use the ``show_axes`` argument to see how
+# each coordinate frame is aligned (shortest arrow is right/X,
+# medium is forward/anderior/Y, longest is up/superior/Z, i.e., RAS
```
Comment (jona-sassenhagen) [inline tutorials/plot_source_alignment.py]: anderior -> anterior

File: mne/viz/_3d.py [hunk 4]
```diff
@@ -481,11 +482,25 @@ def plot_alignment(info, trans=None, subject=None, subjects_dir=None,
...
     fig : mayavi figure object | None
         Mayavi Scene (instance of mlab.Figure) in which to plot the alignment.
         If ``None``, creates a new 600x600 pixel figure with black background.
 
         .. versionadded:: 0.16
+    interaction : str
+        Can be "trackball" (default) or "terrain", i.e. a turntable-style
+        camera.
+
+        .. versionadded:: 0.16
```
Comment (jona-sassenhagen) [inline mne/viz/_3d.py]: If there's only two options, might also be worth having a boolean choice (`trackball_interaction=bool`)

Comment (larsoner) [reply mne/viz/_3d.py]: This mirrors the PySurfer interface, IIRC we wanted the `str` to leave open future possibilities (while still being as readable/just a touch less usable)

**Comment by mmagnuski** [comment]
I played with the notebook, looks very nice, the only problem I have is that I get this error when trying to open the GUI (but I remember having it before this PR, so I can open a separate issue about it):
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-12-8dc6699564c7> in <module>()
----> 1 mne.gui.coregistration(subject='sample', subjects_dir=subjects_dir)

c:\proj\src\mne-python\mne\gui\__init__.py in coregistration(tabbed, split, scene_width, inst, subject, subjects_dir, guess_mri_subject, scene_height, head_opacity, head_high_res, trans, scrollable, verbose)

c:\proj\src\mne-python\mne\utils.py in verbose(function, *args, **kwargs)
    726         with use_log_level(verbose_level):
    727             return function(*args, **kwargs)
--> 728     return function(*args, **kwargs)
    729 
    730 

c:\proj\src\mne-python\mne\gui\__init__.py in coregistration(tabbed, split, scene_width, inst, subject, subjects_dir, guess_mri_subject, scene_height, head_opacity, head_high_res, trans, scrollable, verbose)
    127     frame = CoregFrame(inst, subject, subjects_dir, guess_mri_subject,
    128                        head_opacity, head_high_res, trans, config)
--> 129     return _initialize_gui(frame, view)
    130 
    131 

c:\proj\src\mne-python\mne\gui\__init__.py in _initialize_gui(frame, view)
     16         return frame.edit_traits(view=view), frame
     17     else:
---> 18         frame.configure_traits(view=view)
     19         return frame
     20 

C:\Anaconda3\envs\pyqt4\lib\site-packages\traits\has_traits.py in configure_traits(self, filename, view, kind, edit, context, handler, id, scrollable, **args)
   2154                 context = self
   2155             rc = toolkit().view_application( context, self.trait_view( view ),
-> 2156                                            kind, handler, id, scrollable, args )
   2157             if rc and (filename is not None):
   2158                 fd = None

C:\Anaconda3\envs\pyqt4\lib\site-packages\traitsui\qt4\toolkit.py in view_application(self, context, view, kind, handler, id, scrollable, args)
    211         from . import view_application
    212         return view_application.view_application( context, view, kind, handler,
--> 213                                                   id, scrollable, args )
    214 
    215     #---------------------------------------------------------------------------

C:\Anaconda3\envs\pyqt4\lib\site-packages\traitsui\qt4\view_application.py in view_application(context, view, kind, handler, id, scrollable, args)
     88                   id         = id,
     89                   scrollable = scrollable,
---> 90                   args       = args )
     91 
     92     # If the UI has not been closed yet, we need to keep a reference to

C:\Anaconda3\envs\pyqt4\lib\site-packages\traitsui\view.py in ui(self, context, parent, kind, view_elements, handler, id, scrollable, args)
    431             kind = self.kind
    432 
--> 433         ui.ui( parent, kind )
    434 
    435         return ui

C:\Anaconda3\envs\pyqt4\lib\site-packages\traitsui\ui.py in ui(self, parent, kind)
    220                                    dispatch = 'ui' )
    221         self.rebuild = getattr( toolkit(), 'ui_' + kind )
--> 222         self.rebuild( self, parent )
    223 
    224     #---------------------------------------------------------------------------

C:\Anaconda3\envs\pyqt4\lib\site-packages\traitsui\qt4\toolkit.py in ui_live(self, ui, parent)
    153         """
    154         from . import ui_live
--> 155         ui_live.ui_live( ui, parent )
    156 
    157     def ui_modal ( self, ui, parent ):

C:\Anaconda3\envs\pyqt4\lib\site-packages\traitsui\qt4\ui_live.py in ui_live(ui, parent)
     37     """Creates a live, non-modal PyQt user interface for a specified UI object.
     38     """
---> 39     _ui_dialog(ui, parent, BaseDialog.NONMODAL)
     40 
     41 def ui_livemodal(ui, parent):

C:\Anaconda3\envs\pyqt4\lib\site-packages\traitsui\qt4\ui_live.py in _ui_dialog(ui, parent, style)
     57         ui.owner = _LiveWindow()
     58 
---> 59     BaseDialog.display_ui(ui, parent, style)
     60 
     61 

C:\Anaconda3\envs\pyqt4\lib\site-packages\traitsui\qt4\ui_base.py in display_ui(ui, parent, style)
    480         """Display the UI."""
    481 
--> 482         ui.owner.init(ui, parent, style)
    483         ui.control = ui.owner.control
    484         ui.control._parent = parent

C:\Anaconda3\envs\pyqt4\lib\site-packages\traitsui\qt4\ui_live.py in init(self, ui, parent, style)
    175             bbox = None
    176 
--> 177         self.add_contents(panel(ui), bbox)
    178 
    179     def close(self, rc=True):

C:\Anaconda3\envs\pyqt4\lib\site-packages\traitsui\qt4\ui_panel.py in panel(ui)
    253         panel = None
    254     if nr_groups == 1:
--> 255         panel = _GroupPanel(content[0], ui).control
    256     elif nr_groups > 1:
    257         panel = QtGui.QTabWidget()

C:\Anaconda3\envs\pyqt4\lib\site-packages\traitsui\qt4\ui_panel.py in __init__(self, group, ui, suppress_label)
    543             self._setup_editor(group, editor)
    544 
--> 545             self._add_splitter_items(content, splitter)
    546 
    547         elif group.layout in ('tabbed', 'fold'):

C:\Anaconda3\envs\pyqt4\lib\site-packages\traitsui\qt4\ui_panel.py in _add_splitter_items(self, content, splitter)
    621             # Get a panel for the Item or Group.
    622             if isinstance(item, Group):
--> 623                 panel = _GroupPanel(item, self.ui, suppress_label=True).control
    624             else:
    625                 panel = self._add_items([item])

C:\Anaconda3\envs\pyqt4\lib\site-packages\traitsui\qt4\ui_panel.py in __init__(self, group, ui, suppress_label)
    588 
    589             if isinstance(content[0], Group):
--> 590                 layout = self._add_groups(content, inner)
    591             else:
    592                 layout = self._add_items(content, inner)

C:\Anaconda3\envs\pyqt4\lib\site-packages\traitsui\qt4\ui_panel.py in _add_groups(self, content, outer)
    667         # Process each group.
    668         for subgroup in content:
--> 669             panel = _GroupPanel(subgroup, self.ui).control
    670 
    671             if isinstance(panel, QtGui.QWidget):

C:\Anaconda3\envs\pyqt4\lib\site-packages\traitsui\qt4\ui_panel.py in __init__(self, group, ui, suppress_label)
    588 
    589             if isinstance(content[0], Group):
--> 590                 layout = self._add_groups(content, inner)
    591             else:
    592                 layout = self._add_items(content, inner)

C:\Anaconda3\envs\pyqt4\lib\site-packages\traitsui\qt4\ui_panel.py in _add_groups(self, content, outer)
    667         # Process each group.
    668         for subgroup in content:
--> 669             panel = _GroupPanel(subgroup, self.ui).control
    670 
    671             if isinstance(panel, QtGui.QWidget):

C:\Anaconda3\envs\pyqt4\lib\site-packages\traitsui\qt4\ui_panel.py in __init__(self, group, ui, suppress_label)
    590                 layout = self._add_groups(content, inner)
    591             else:
--> 592                 layout = self._add_items(content, inner)
    593             layout.setAlignment(QtCore.Qt.AlignLeft | QtCore.Qt.AlignTop)
    594 

C:\Anaconda3\envs\pyqt4\lib\site-packages\traitsui\qt4\ui_panel.py in _add_items(self, content, outer)
    858             # that it is not None (otherwise the main TraitsUI code can change
    859             # the "kind" of the created UI object).
--> 860             editor.prepare(inner)
    861             control = editor.control
    862 

C:\Anaconda3\envs\pyqt4\lib\site-packages\traitsui\editor.py in prepare(self, parent)
    154             self.context_object.on_trait_change( self._update_editor, name,
    155                                                  dispatch = 'ui' )
--> 156         self.init( parent )
    157         self._sync_values()
    158         self.update_editor()

C:\Anaconda3\envs\pyqt4\lib\site-packages\traitsui\qt4\enum_editor.py in init(self, parent)
    367                                self.update_object)
    368 
--> 369         self.rebuild_editor()
    370 
    371     #---------------------------------------------------------------------------

C:\Anaconda3\envs\pyqt4\lib\site-packages\traitsui\qt4\enum_editor.py in rebuild_editor(self)
    431             for j in range( cols ):
    432                 if n > 0:
--> 433                     name = names[index]
    434                     rb = self.create_button(name)
    435                     rb.value = mapping[name]

TypeError: list indices must be integers or slices, not float
```

File: tutorials/plot_source_alignment.py [hunk 5]
```diff
@@ -135,3 +203,5 @@
 # to warp a subject (usually ``fsaverage``) to subject digitization data, see
 # `these slides
 # <http://www.slideshare.net/mne-python/mnepython-scale-mri>`_.
+#
+# .. _trans_matrices: https://en.wikipedia.org/wiki/False_discovery_rate
```
Comment (mmagnuski) [inline tutorials/plot_source_alignment.py]: does not seem to be the correct link, this one belongs to the beautiful Statistical Inference docs :)

**Comment by larsoner** [comment]
That looks unrelated, yes please open another issue. FYI I just pushed some updates that make things clearer based on @wmvanvliet's and @jona-sassenhagen's comments, too

https://6460-1301584-gh.circle-artifacts.com/0/tmp/circle-artifacts.Ne77gZS/html/auto_tutorials/plot_source_alignment.html

File: tutorials/plot_source_alignment.py (hunk 5 above)
Comment (jona-sassenhagen) [reply tutorials/plot_source_alignment.py]: (Agree, the stats docs are great.)

**Comment by jona-sassenhagen** [comment]
The section explaining the 3 transforms looks very good now.

**Comment by agramfort** [comment]
looks great ! good to go from my end

**Comment by larsoner** [comment]
Okay it seems like everyone is happy. @wmvanvliet feel free to look again and open a PR with edits or issue for ideas

**Comment by wmvanvliet** [comment]
I'm too lazy/busy :) The tutorial is cool as it is.

**Pull request merged**